src/routes/events/containers/EventsPage.js

```diff
@@ -26,6 +29,43 @@ class EventsPage extends React.Component {
     return `/${this.props.location}/${this.props.language}/events`
   }
 
+  mapLanguageToUrl = (language, id) =>
+    id ? `/${this.props.location}/${language}/events/${id}`
+      : `/${this.props.location}/${language}/events`
+
+  componentDidMount () {
+    if (this.props.path && this.props.events) {
```

Comment:
Can be simplified to:
```
    let availableLanguages = {}

    if (this.props.path && this.props.events) {
      // specific event
      const event = this.props.events.find(
        (event) => event.id.toString() === this.props.path.replace('/', '') // Works because we can not have nested events
      )
      availableLanguages = event.availableLanguages
    }

    this.props.dispatch(setLanguageChangeUrls(this.mapLanguageToUrl, this.props.languages, availableLanguages))
```

Reply:
done